File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/InventoryAssetResponse.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;
import java.util.UUID;

/**
 * response to RequestInventoryAsset
 * lluuid will be null if agentid in the request above cannot read asset
 *
 * <p>Template: {@code InventoryAssetResponse Low 283 Trusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
public class InventoryAssetResponse extends SLMessage {
    public QueryData QueryData_Field;

    /** Block QueryData, Single. */
    public static class QueryData {
        public UUID AssetID; // LLUUID
        public boolean IsReadable; // BOOL
        public UUID QueryID; // LLUUID
    }

    public InventoryAssetResponse() {
        this.zeroCoded = false;
        this.QueryData_Field = new QueryData();
    }

    @Override
    public int CalcPayloadSize() {
        return 37;
    }

    @Override
    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleInventoryAssetResponse(this);
    }

    @Override
    public void PackPayload(ByteBuffer byteBuffer) {
        // Message number: Low 283 (InventoryAssetResponse).
        byteBuffer.putShort((short) 0xFFFF);
        byteBuffer.put((byte) 0x01);
        byteBuffer.put((byte) 0x1B);
        packUUID(byteBuffer, this.QueryData_Field.QueryID);
        packUUID(byteBuffer, this.QueryData_Field.AssetID);
        packBoolean(byteBuffer, this.QueryData_Field.IsReadable);
    }

    @Override
    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.QueryData_Field.QueryID = unpackUUID(byteBuffer);
        this.QueryData_Field.AssetID = unpackUUID(byteBuffer);
        this.QueryData_Field.IsReadable = unpackBoolean(byteBuffer);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/InventoryAssetResponse.kt
package com.lumiyaviewer.lumiya.slproto.messages

import com.lumiyaviewer.lumiya.slproto.SLMessage
import java.nio.ByteBuffer
import java.util.UUID

/**
 * response to RequestInventoryAsset
 * lluuid will be null if agentid in the request above cannot read asset
 *
 * <p>Template: {@code InventoryAssetResponse Low 283 Trusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
open class InventoryAssetResponse : SLMessage() {
    @JvmField var QueryData_Field: QueryData = QueryData()

    /** Block QueryData, Single. */
    open class QueryData {
        @JvmField var AssetID: UUID? = null
        @JvmField var IsReadable: Boolean = false
        @JvmField var QueryID: UUID? = null
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return 37
    }

    override fun Handle(messageHandler: SLMessageHandler) {
        messageHandler.HandleInventoryAssetResponse(this)
    }

    override fun PackPayload(byteBuffer: ByteBuffer) {
        // Message number: Low 283 (InventoryAssetResponse).
        byteBuffer.putShort(0xFFFF.toShort())
        byteBuffer.put((0x01).toByte())
        byteBuffer.put((0x1B).toByte())
        packUUID(byteBuffer, QueryData_Field.QueryID)
        packUUID(byteBuffer, QueryData_Field.AssetID)
        packBoolean(byteBuffer, QueryData_Field.IsReadable)
    }

    override fun UnpackPayload(byteBuffer: ByteBuffer) {
        QueryData_Field.QueryID = unpackUUID(byteBuffer)
        QueryData_Field.AssetID = unpackUUID(byteBuffer)
        QueryData_Field.IsReadable = unpackBoolean(byteBuffer)
    }
}
